src/axolotl/core/trainer_builder.py

```diff
@@ -410,10 +426,12 @@ def __init__(
         *_args,
         bench_data_collator=None,
         eval_data_collator=None,
+        dataset_tag_names=None,
```

Comment:
it is used for both AxolotlTrainer and AxolotlDPOTrainer within `def push_to_hub` via `_sanitize_kwargs_for_ds_tagging`

Reply:
I'm not fond of introducing arguments that don't match the original Trainer signature for this case. The call to `_sanitize_kwargs_for_ds_tagging` should handle the case where the attribute isn't available.  In this particular case, AxolotlTrainer would never have `dataset_tag_names` set since it's not part of the defined interface of Trainer, so calling `_sanitize_kwargs_for_ds_tagging` with `self.dataset_tag_names` is redundant, and you never use `_sanitize_kwargs_for_ds_tagging` in the AxolotlDPOTrainer, so I don't really understand the use case of `_sanitize_kwargs_for_ds_tagging`

Reply:
Thanks for the advice. While trl trainer has an override for `def create_model_card` over the base transformers trainer, it uses the same `def push_to_hub` inherited from the base transformers trainer. I changed `dataset_tag_names` to `dataset_tags` to match `def push_to_hub`

Reply:
@winglian Pls let me know what you think. Thanks!